marathon/src/main/scala/io/buoyant/marathon/v2/Api.scala

```diff
@@ -146,33 +162,40 @@ object Api {
   private[this] def toAppIds(appsRsp: AppsRsp): Api.AppIds = {
     appsRsp.apps match {
       case Some(apps) =>
-        apps.collect { case App(Some(id), _, _) => Path.read(id.toLowerCase) }.toSet
+        apps.collect { case App(Some(id), _, _, _) => Path.read(id.toLowerCase) }.toSet
       case None => Set.empty
     }
   }
 
   private[v2] def toAddresses(appRsp: AppRsp, useHealthCheck: Boolean): Set[Address] =
     appRsp.app match {
-      case Some(App(_, discoveryPort(port), Some(tasks))) =>
-        tasks.collect {
-          case ipAddress(host) if !useHealthCheck => Address(host, port)
-          case healthyIpAddress(host) => Address(host, port)
-        }.toSet
-      case Some(App(_, _, Some(tasks))) =>
-        tasks.collect {
-          case hostPort(host, port) if !useHealthCheck => Address(host, port)
-          case healthyHostPort(host, port) => Address(host, port)
-        }.toSet
+      case Some(App(_, discoveryPort(port), healthCheckCount(count), Some(tasks))) =>
```

Comment:
I think we can leave this method largely unchanged if we incorporate the number of health checks into the `Task` case class.

Reply:
@rogoman correct me if I'm wrong but I believe that Task is a class that matches the Marathon API and API responses are automatically deserialized to it.  This means that we can't add the health check count to that class.

Reply:
@adleong yes. Adding the number of health checks to the original `Task` case class was the easiest to implement, but I worried that it's going to be confusing for anyone reading the code and trying to find a match with what can be found in the Marathon API.